### PR TITLE
OTel metrics for Microprofile Telemetry 2.0

### PR DESCRIPTION
**Pull request by brunobat**
Provides OpenTelemetry metrics according to https://github.com/eclipse/microprofile-telemetry/blob/2.0/spec/src/main/asciidoc/metrics.adoc

### CODE REVIEW

**Comment by brunobat** [comment]
Will include the TCK updates as well... Set to draft until then.

**Comment by Ladicek** [comment]
Off topic (but not all that much I guess): are you planning to support `MetricsCapabilityBuildItem`? (And the related SPIs too, probably?)

**Comment by brunobat** [comment]
> Off topic (but not all that much I guess): are you planning to support `MetricsCapabilityBuildItem`? (And the related SPIs too, probably?)

I'll decide that after evaluating the future Micrometer to OpenTelemetry bridge. 
It it goes well, there might be no more automatic metrics for OTel.

**Comment by brunobat** [comment]
https://github.com/quarkusio/quarkus/pull/43752 is needed for the native image tests to pass because it includes an OTel side instrumentation fix.

**Comment by brunobat** [comment]
There are 2 implementations for this failing metric ThreadCountMetric. One for Java 17:
[io.opentelemetry.instrumentation.runtimemetrics.java17.internal.threads.ThreadCountHandler#METRIC_DESCRIPTION
](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/39f4891b212e1e5b80cc0545db7a29103ed3c468/instrumentation/runtime-telemetry/runtime-telemetry-java17/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java17/internal/threads/ThreadCountHandler.java#L26)Another one for Java 8:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/745d880c06676a28d75c5567aa7bd5ad68c5da2d/instrumentation/runtime-telemetry/runtime-telemetry-java8/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java8/Threads.java#L69

They both use different descriptions and the TCK requires the Java 8 flavour:
https://github.com/eclipse/microprofile-telemetry/blob/88735b74308d123a324b66cc9c4a132f980cb541/tck/metrics/src/main/java/org/eclipse/microprofile/telemetry/metrics/tck/jvm/JvmThreadTest.java#L59

**Review by gsmet — COMMENTED** [review]
The changes look good to me except for some minor nitpicking in the doc - but I might have misunderstood the sentence.

As for the failing test, should we raise an issue in the TCK and for now ignore this test?

**Comment by brunobat** [comment]
Created https://github.com/eclipse/microprofile-telemetry/issues/241

**Comment by radcortez** [comment]
> Created [eclipse/microprofile-telemetry#241](https://github.com/eclipse/microprofile-telemetry/issues/241)

You can also exclude the failing test (because of the challenge) so we can proceed with the PR. Check:
https://github.com/smallrye/smallrye-opentelemetry/blob/a750353e55ccede3fe498bbc693be21621dac7a5/testsuite/tck/pom.xml#L97-L105

File: docs/src/main/asciidoc/opentelemetry-metrics.adoc [hunk 1]
```diff
@@ -529,9 +529,23 @@ We only show 1 of 3 exemplars for brevity.
 === Resource
 See the main xref:opentelemetry.adoc#resource[OpenTelemetry Guide resources] section.
 
-== Additional instrumentation
+== Automatic instrumentation
+
+We provide automatic instrumentation JVM Metrics and HTTP server requests according to the https://github.com/eclipse/microprofile-telemetry/blob/2.0/spec/src/main/asciidoc/metrics.adoc[Microprofile Metrics 2.0 specification].
```
[the remark this answers is not in the document]
Comment (brunobat) [reply docs/src/main/asciidoc/opentelemetry-metrics.adoc]: fixed the docs

**Comment by gsmet** [comment]
@radcortez I will let you review this one. I could be convinced to shoe it in 3.16.0 post CR1 if we think it has high value for 3.16.

**Review by radcortez — APPROVED** [review]

**Pull request merged**